### PR TITLE
Cvsl 1668 hard stop standard licence confirm create page and appointment with page

### PR DESCRIPTION
**Pull request by nishanthmahasamudram**

### CODE REVIEW

File: server/routes/creatingLicences/index.ts [hunk 1]
```diff
@@ -41,18 +42,18 @@ export default function Index({
...
     router.get(
       routePrefix(path),
-      roleCheckMiddleware(['ROLE_LICENCE_RO']),
+      roleCheckMiddleware([role, 'ROLE_LICENCE_RO']),
       fetchLicence(licenceService),
       asyncMiddleware(handler)
     )
 
-  const post = (path: string, handler: RequestHandler, type?: new () => object) =>
+  const post = (path: string, handler: RequestHandler, type?: new () => object, role?: AuthRole) =>
```
Comment (andrewrlee) [inline server/routes/creatingLicences/index.ts:53]: not sure about this, presumably these changes aren't being used?

File: server/services/licenceService.test.ts [hunk 2]
```diff
@@ -59,8 +59,11 @@ describe('Licence Service', () => {
 
   describe('Create Licence', () => {
     it('Should create a licence in the backend API', async () => {
-      await licenceService.createLicence('ABC1234', user)
-      expect(licenceApiClient.createLicence).toHaveBeenCalledWith({ nomsId: 'ABC1234' }, user)
+      await licenceService.createLicence({ nomsId: 'ABC1234', type: 'CRD' }, user)
+      expect(licenceApiClient.createLicence).toHaveBeenCalledWith({ nomsId: 'ABC1234', type: 'CRD' }, user)
+
+      await licenceService.createLicence({ nomsId: 'ABC1235', type: 'HARD_STOP' }, user)
```
Comment (andrewrlee) [inline server/services/licenceService.test.ts:65]: would usually split these into separate tests

File: server/services/licenceService.test.ts [hunk 3]
```diff
@@ -70,8 +73,32 @@ describe('Licence Service', () => {
...
+      {
+        contactName: '',
+        appointmentPersonType: 'DUTY_OFFICER',
+      },
+      user
+    )
+    expect(licenceApiClient.updateAppointmentPerson).toBeCalledWith(
+      '1',
+      { appointmentPerson: '', appointmentPersonType: 'DUTY_OFFICER' },
+      user
```
Comment (andrewrlee) [inline server/services/licenceService.test.ts:100]: would usually split these into separate tests

File: server/views/partials/viewCases.njk [hunk 4]
```diff
@@ -5,12 +5,16 @@
...
-            {% set link = "/licence/view/id/" + licenceId + "/show" %}
-            {% if versionOfId and licenceStatus === 'SUBMITTED' %}
-                {% set link = link + "?lastApprovedVersion=" + versionOfId %}
+            {% if isHardStop %}
+                {% set link = "/licence/hardstop/create/id/" + licenceId + "/check-your-answers" if licenceId else "/licence/hardstop/create/nomisId/" + prisonerNumber + "/confirm"%}
+            {% else %}
+                {% set link = "/licence/view/id/" + licenceId + "/show" %}
+                {% if versionOfId and licenceStatus === 'SUBMITTED' %}
+                    {% set link = link + "?lastApprovedVersion=" + versionOfId %}
+                {% endif %}
```
Comment (andrewrlee) [inline server/views/partials/viewCases.njk:17]: this is not part of that ticket.. we're looking in to handling this outside of the nunjucks as its quite complicated.

This ticket just covers the create journey flow without the entry point - so the page would exist but you'd have to hit it via the url directly

Comment (nishanthmahasamudram) [reply server/views/partials/viewCases.njk]: I have removed this code.
We can access licence creation flow only using url for now.

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**